src/main/webapp/js/common/react/Navbar.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Returns a React navbar component.
+ * @param {{}} props The properties of the navbar.
+ * @return Returns the navbar.
+ */
+function Navbar(props) {
+  const urls = [
+    {
+      href: "/",
+      name: "Home"
+    },
+    {
+      href: "/projects.html",
+      name: "Projects"
+    }
+  ];
+
+  return (
+    <nav className="navbar navbar-expand-lg navbar-light bg-light">
+      <a className="navbar-brand" href="#">Open Sesame</a>
+      <button 
+          className="navbar-toggler" 
+          type="button" 
+          data-toggle="collapse" 
+          data-target="#navbarSupportedContent" 
+          aria-controls="navbarSupportedContent" 
+          aria-expanded="false" 
+          aria-label="Toggle navigation">
+        <span className="navbar-toggler-icon"></span>
+      </button>
+      <div className="collapse navbar-collapse" id="navbarSupportedContent">
+        <ul className="navbar-nav ml-auto">
+          {urls.map((url, i) => 
+              <NavbarLink key={i} href={url.href} name={url.name} />)}
+        </ul>
+      </div>
+    </nav>
+  );
+}
+
+/**
+ * Returns a React navbar link component.
+ * @param {{href: string, name: string}} The properties of the navbar link.
+ * @return Returns the navbar link.
+ */
+function NavbarLink(props) {
+  let classes = "nav-item";
+  if (props.href === window.location.pathname) {
+    classes += " active";
+  }
+  
+  return (
+    <div className={classes}>
+      <a className="nav-link" href={props.href}>{props.name}</a>
+    </div>
+  );
+}
+
+const navbarContainer = document.getElementById("navbar-container");
+ReactDOM.render(<Navbar />, navbarContainer);
```

Comment:
Just a Q: wondering how this rendering gets called? (doesn't immediately appear to be part of a function)

Reply:
This is called when the script is loaded as a part of the global scope. And if you're wondering about motivation, it allows the navbar to be loaded when the script is included, and avoids having to call any loading functions. I'd like to hear Bill or Joe's opinion on if this is considered smelly code tho.